### ISSUE TITLE
Duplicate Scott L Wing in prod

### ISSUE THREAD

**Issue by ecurrano** [paleobot/pbot-dev#297]
Prod seems to have no duplicate people, except Scott L Wing and Scott L. Wing.

I think I had Scott register before we had that all figured out, and he created Scott L. Wing.

**Comment by aazaff** [comment]
Is it possible this one slipped through the manual merge @NoisyFlowers did before because of the period?

**Comment by ecurrano** [comment]
We've now also got a Nathan A Jud and Nathan A. Jud. Nathan just registered for PBot, and there was already a user node for him and these didn't get connected. Note that Nathan registered under a different email address than what had been listed for him.

This raises the larger question on how the search is done and making sure we are properly educating people on registering and uniting person nodes.

**Comment by aazaff** [comment]
For convenience, the last time @NoisyFlowers fixed this was in #233

**Comment by NoisyFlowers** [comment]
Scott L Wing
74347fd3-8bd6-4936-b705-600dc045ef97 is correct, is registered, has two ENTERED_BY, and two MEMBER_OF
cee14744-8722-4595-9e27-bc0918c31fe7 is unregistered, has many HAS_IDENTIFIED, three _HAS_IDENTIFIED, six AUTHORED_BY, one MEMBER_OF other than public

cypher executed:
```
		MATCH
			(p_right:Person {pbotID:"74347fd3-8bd6-4936-b705-600dc045ef97"}),
			(p_wrong:Person {pbotID: "cee14744-8722-4595-9e27-bc0918c31fe7"})<-[ab_rel:AUTHORED_BY]-()
		CALL apoc.refactor.to(ab_rel, p_right)
		YIELD input, output
		RETURN input, output;	
	
		MATCH
			(p_right:Person {pbotID:"74347fd3-8bd6-4936-b705-600dc045ef97"}),
			(p_wrong:Person {pbotID: "cee14744-8722-4595-9e27-bc0918c31fe7"})-[ab_rel:HAS_IDENTIFIED]->()
		CALL apoc.refactor.from(ab_rel, p_right)
		YIELD input, output
		RETURN input, output;	
		
		MATCH
			(p_right:Person {pbotID:"74347fd3-8bd6-4936-b705-600dc045ef97"}),
			(p_wrong:Person {pbotID: "cee14744-8722-4595-9e27-bc0918c31fe7"})-[ab_rel:_HAS_IDENTIFIED]->()
		CALL apoc.refactor.from(ab_rel, p_right)
		YIELD input, output
		RETURN input, output;					
		
		MATCH
			(p_right:Person {pbotID:"74347fd3-8bd6-4936-b705-600dc045ef97"}),
			(p_wrong:Person {pbotID: "cee14744-8722-4595-9e27-bc0918c31fe7"})-[ab_rel:MEMBER_OF]->({pbotID:"5033069a-43da-470c-babb-e2cc8f011883"})
		CALL apoc.refactor.from(ab_rel, p_right)
		YIELD input, output
		RETURN input, output;			
		
		MATCH
			(p_wrong:Person {pbotID: "cee14744-8722-4595-9e27-bc0918c31fe7"})
		DETACH DELETE
			p_wrong		
```			
			
Nathan Jud
cfd50257-30c0-4312-b64a-fedf3a2421e2 is correct, is registered
582cb38b-ddee-45d3-ac08-d81e84274526 is unregistered, has one AUTHORED_BY

cypher executed:
```
		MATCH
			(p_right:Person {pbotID:"cfd50257-30c0-4312-b64a-fedf3a2421e2"}),
			(p_wrong:Person {pbotID: "582cb38b-ddee-45d3-ac08-d81e84274526"})<-[ab_rel:AUTHORED_BY]-()
		CALL apoc.refactor.to(ab_rel, p_right)
		YIELD input, output
		RETURN input, output;	

		MATCH
			(p_wrong:Person {pbotID: "582cb38b-ddee-45d3-ac08-d81e84274526"})
		DETACH DELETE
			p_wrong		
```

**Comment by NoisyFlowers** [comment]
Since we use email address as the primary unique marker and do not yet provide a way to edit account data, using a different email address to register will always result in a new Person node.

I think we should consider not allowing entry of email address when a Person node is created outside of the registration process.

**Comment by aazaff** [comment]
Yes! Absolutely that would be at least one helpful fix, but overall I think this was a design flaw on my part of having person and account concept be linked

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: NoisyFlowers ***@***.***>
Sent: Wednesday, September 11, 2024 4:22:28 PM
To: paleobot/pbot-dev ***@***.***>
Cc: Zaffos, Andrew - (azaffos) ***@***.***>; Comment ***@***.***>
Subject: [EXT] Re: [paleobot/pbot-dev] Duplicate Scott L Wing in prod (Issue #297)


External Email

________________________________
Since we use email address as the primary unique marker and do not yet provide a way to edit account data, using a different email address to register will always result in a new Person node.

I think we should consider not allowing entry of email address when a Person node is created outside of the registration process.
